Machine.Specifications.Should/ShouldExtensionMethods.cs

```diff
@@ -718,5 +719,32 @@ private static bool IsReferenceTypeNotNullOrString(object obj)
         {
             return obj != null && obj.GetType().IsClass && !(obj is string);
         }
+
+        // Required for ShouldBeLikeInternal(), unfortunately we do not have Sytem.Tuple<T1,T2> in .NET < 4.0
```

Comment:
Note: `Tuple` again would use equality instead of referential equality (see https://github.com/machine/machine.specifications.should/pull/18/commits/2f9f3d8d94c4787095d371586c38054b3f9e5fa3#r56924235), so it's good to use this "ReferentialEqualityObjAndExpected" tuple type. 

But: Don't call .GetHashCode(), instead use [RuntimeHelpers.GetHashCode](https://msdn.microsoft.com/en-us/library/11tbk3h9%28v=vs.110%29.aspx), otherwise we break the euality contract of `SimpleTuple`.